File: src/main/java/models/UserInfo.java
package models;

import nosqlite.annotations.Document;
import nosqlite.annotations.Id;

@Document
public class UserInfo {

    @Id
    private String id;
    private String userId;
    private String firstName;
    private String lastName;
    private String streetAddress;
    private String zipCode;
    private String city;
    private String country;
    private String phoneNumber;

    public UserInfo() {}

    public UserInfo(String userId, String firstName, String lastName, String streetAddress, String zipCode, String city, String country, String phoneNumber) {
        this.userId = userId;
        this.firstName = firstName;
        this.lastName = lastName;
        this.streetAddress = streetAddress;
        this.zipCode = zipCode;
        this.city = city;
        this.country = country;
        this.phoneNumber = phoneNumber;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getStreetAddress() {
        return streetAddress;
    }

    public void setStreetAddress(String streetAddress) {
        this.streetAddress = streetAddress;
    }

    public String getZipCode() {
        return zipCode;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getCountry() {
        return country;
    }

    @Override
    public String toString() {
        return "UserInfo{" +
                "id='" + id + '\'' +
                ", userId='" + userId + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", streetAddress='" + streetAddress + '\'' +
                ", zipCode='" + zipCode + '\'' +
                ", city='" + city + '\'' +
                ", country='" + country + '\'' +
                ", phoneNumber='" + phoneNumber + '\'' +
                '}';
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

}

File: frontend/src/components/UserInfo.js
export default class UserInfo {
  constructor(
    userId,
    firstName,
    lastName,
    streetAddress,
    zipCode,
    city,
    country,
    phoneNumber,
    newsLetter
  ) {
    this.userId = userId;
    this.firstName = firstName;
    this.lastName = lastName;
    this.streetAddress = streetAddress;
    this.zipCode = zipCode;
    this.city = city;
    this.country = country;
    this.phoneNumber = phoneNumber;
    this.newsLetter = newsLetter;
  }
  getUserId() {
    return this.userId;
  }
  getFirstName() {
    return this.firstName;
  }
  setFirstName(newFirstName) {
    this.firstName = newFirstName;
  }
  getLastName() {
    return this.lastName;
  }
  setLastName(newLastName) {
    this.lastName = newLastName;
  }
  getStreetAddress() {
    return this.streetAddress;
  }
  setStreetAddress(newStreetAddress) {
    this.streetAddress = newStreetAddress;
  }
  getZipCode() {
    return this.zipCode;
  }
  setZipCode(newZipCode) {
    this.zipCode = newZipCode;
  }
  getCity() {
    return this.city;
  }
  setCity(newCity) {
    this.city = newCity;
  }
  getCountry() {
    return this.country;
  }
  setCountry(newCountry) {
    this.country = newCountry;
  }
  getPhoneNumber() {
    return this.phoneNumber;
  }
  setPhoneNumber(newPhoneNumber) {
    this.phoneNumber = newPhoneNumber;
  }
  getNewsLetter() {
    return this.newsLetter;
  }
  setNewsLetter(wantsNewsLetter) {
    this.newsLetter = wantsNewsLetter;
  }
}
